.gitlab/kitchen_testing/new-e2e_testing/windows.yml

```diff
@@ -165,7 +165,6 @@ new-e2e-windows-agent-msi-upgrade-windows-server-a7-x86_64:
     - !reference [.except_main_or_release_branch]
     - !reference [.except_windows_installer_changes]
     - !reference [.on_default_new_e2e_tests]
-    # must be last since it ends with when: on_success
     - !reference [.except_deploy]
```

Comment:
Note for future readers: this change does not actually modify the behavior of the job, even though it removes a `- when: on_success` from the list of rules.

The previous reference on this job, `.on_default_new_e2e_tests`, defines the following:
```
    - if: $RUN_E2E_TESTS == "off"
      when: never
    - if: ($CI_COMMIT_BRANCH == "main"  || $DEPLOY_AGENT == "true" || $RUN_E2E_TESTS
        == "on" || $DDR_WORKFLOW_ID != null) && $RUN_E2E_TESTS != "off"
    - if: $RUN_E2E_TESTS == "auto"
      variables:
        E2E_OSVERS: $E2E_BRANCH_OSVERS
```
Ie:
- when `$RUN_E2E_TESTS` is`off`, do not run,
- if `$RUN_E2E_TESTS` is `on` or `auto`, run.

Assuming that { `on`, `auto`, `off` } are the only valid values for `$RUN_E2E_TESTS` - which is true today -, then the `- when: on_success` section is never reached, and removing it has no impact.

A corollary is that the `.except_deploy` block is also never reached. To have an effect, it should be put above `.on_default_new_e2e_tests` - but I'd do this in a separate PR, since this is not in the scope of the current PR.